pkg/stis/calstis/cs0/cs0.c

```diff
@@ -130,41 +144,46 @@ int main (int argc, char **argv) {
 		    } else {
 			printf ("Unrecognized option %s\n", argv[i]);
 			printSyntax();
-			FreeNames (rawlist, wavlist, outlist,
-				rawfile, wavfile, outroot);
+			freeOnExit(&ptrReg);
 			exit (ERROR_RETURN);
 		    }
 		}
 	    } else if (rawlist[0] == '\0') {
-		free (rawlist);
+		freePtr(&ptrReg, rawlist);
 		if ((rawlist = calloc (strlen(argv[i])+1, sizeof(char)))
 			== NULL) {
 		    printf ("ERROR:  Out of memory.\n");
+		    freeOnExit(&ptrReg);
 		    exit (ERROR_RETURN);
 		}
+		addPtr(&ptrReg, rawlist, &free);
 		strcpy (rawlist, argv[i]);
 	    } else if (outlist[0] == '\0') {
-		free (outlist);
+		freePtr(&ptrReg, outlist);
 		if ((outlist = calloc (strlen(argv[i])+1, sizeof(char)))
 			== NULL) {
 		    printf ("ERROR:  Out of memory.\n");
 		    exit (ERROR_RETURN);
 		}
+		addPtr(&ptrReg, rawlist, &free);
```

Comment:
Did you mean this to be outlist instead of rawlist?

Reply:
Yes, good catch, thanks.